src/libzerocoin/bignum.h

```diff
@@ -708,3 +717,487 @@ inline bool operator>(const CBigNum& a, const CBigNum& b)  { return (BN_cmp(a.bn
 inline std::ostream& operator<<(std::ostream &strm, const CBigNum &b) { return strm << b.ToString(10); }
 
 #endif
+#if defined(USE_NUM_GMP)
+/** C++ wrapper for BIGNUM (Gmp bignum) */
+class CBigNum
+{
+    mpz_t bn;
+public:
+    CBigNum()
+    {
+        mpz_init(bn);
+    }
+
+    CBigNum(const CBigNum& b)
+    {
+        mpz_init(bn);
+        mpz_set(bn, b.bn);
+    }
+
+    CBigNum& operator=(const CBigNum& b)
+    {
+        mpz_set(bn, b.bn);
+        return (*this);
+    }
+
+    ~CBigNum()
+    {
+        mpz_clear(bn);
+    }
+
+    //CBigNum(char n) is not portable.  Use 'signed char' or 'unsigned char'.
+    CBigNum(signed char n)      { mpz_init(bn); if (n >= 0) mpz_set_ui(bn, n); else mpz_set_si(bn, n); }
+    CBigNum(short n)            { mpz_init(bn); if (n >= 0) mpz_set_ui(bn, n); else mpz_set_si(bn, n); }
+    CBigNum(int n)              { mpz_init(bn); if (n >= 0) mpz_set_ui(bn, n); else mpz_set_si(bn, n); }
+    CBigNum(long n)             { mpz_init(bn); if (n >= 0) mpz_set_ui(bn, n); else mpz_set_si(bn, n); }
+#ifdef __APPLE__
+    CBigNum(int64_t n)            { mpz_init(bn); mpz_set_si(bn, n); }
+#endif
```

Comment:
I really don't think this conditional is even necessary, even further up in the file. Apple typedefs `int64_t` to `long long`, which is available on other systems as well. Couldn't we just use
```cpp
CBigNum(long long n)            { mpz_init(bn); mpz_set_si(bn, n); }
```
and forego the conditional entirely?